pkg/common/cns-lib/volume/manager.go

```diff
@@ -2129,20 +2130,35 @@ func (m *defaultManager) deleteSnapshotWithImprovedIdempotencyCheck(
 	// Handle snapshot operation result
 	deleteSnapshotsOperationRes := deleteSnapshotsTaskResult.GetCnsVolumeOperationResult()
 	if deleteSnapshotsOperationRes.Fault != nil {
-		errMsg := fmt.Sprintf("failed to delete snapshot %s for cns volume %s. fault: %q, opId: %q",
-			snapshotID, volumeID, spew.Sdump(deleteSnapshotsOperationRes.Fault), deleteSnapshotsTaskInfo.ActivationId)
+		err = soap.WrapVimFault(deleteSnapshotsOperationRes.Fault.Fault)
 
-		if m.idempotencyHandlingEnabled {
-			volumeOperationDetails = createRequestDetails(instanceName, "", "", 0,
-				volumeOperationDetails.OperationDetails.TaskInvocationTimestamp, deleteSnapshotTask.Reference().Value,
-				deleteSnapshotsTaskInfo.ActivationId, taskInvocationStatusError, errMsg)
+		isInvalidArgumentError := cnsvsphere.IsInvalidArgumentError(err)
+		var invalidProperty string
+		if isInvalidArgumentError {
+			invalidProperty = deleteSnapshotsOperationRes.Fault.Fault.(*vim25types.InvalidArgument).InvalidProperty
 		}
 
-		return logger.LogNewError(log, errMsg)
-	}
+		// Ignore errors, NotFound and InvalidArgument, in DeleteSnapshot
+		if cnsvsphere.IsNotFoundError(err) {
+			log.Infof("Snapshot %q on volume %q is not found with error %v. Ignore the error for DeleteSnapshot",
+				snapshotID, volumeID, spew.Sdump(deleteSnapshotsOperationRes.Fault))
```

Comment:
Should we log opId for such ignored delete as well?

Reply:
We should say the snapshot is already deleted in the message.

Reply:
>We should say the snapshot is already deleted in the message.

Done @xing-yang 

>Should we log opId for such ignored delete as well?

Yes, we have done that. OpID is logged around L2171.